scripts: add metadata to analyze-styled

JonasBa opened the pull request:
Adds a simple loop to collect data over date intervals and outputs it to tmp dir where it can be analyzed.

JonasBa committed:
scripts: add history functinality to analyze-styled

dry test

remove dry

scripts: add git safety check to analyze-styled history command

Prevents history analysis from running when git tree is dirty to avoid
potential data loss during git checkout operations.

exit safely

break sync loop

force checkout

Fix historical analysis file resolution

- Extract file discovery logic into reusable function
- Modify processCommit to collect files after git checkout
- Update runCurrentAnalysis to accept file list parameter
- Fixes 'file not found' errors in historical analysis by ensuring
  file lists are always relative to the current commit state

wip

wip

store out in memory

outdir

different target

same out

reset numbers

natemoo-re reviewed (approved):
I did read through and nothing jumped out as concerning, but I don't have the full context so consider this a `--yolo` approval 😅

JonasBa commented:
> I did read through and nothing jumped out as concerning, but I don't have the full context so consider this a `--yolo` approval 😅

I'll take it :)

JonasBa committed:
scripts: add metadata to analyze-styled (#99706)

Add a `styled` call site detector so that we can monitor the number of
styled calls as well as the ratio of files with styled calls and those
without
